lib/rspec/core/shared_example_group.rb

```diff
@@ -213,16 +213,35 @@ def valid_name?(candidate)
 
         def warn_if_key_taken(context, key, new_block)
           existing_module = shared_example_groups[context][key]
-
           return unless existing_module
 
-          RSpec.warn_with <<-WARNING.gsub(/^ +\|/, ''), :call_site => nil
-            |WARNING: Shared example group '#{key}' has been previously defined at:
-            |  #{formatted_location existing_module.definition}
-            |...and you are now defining it at:
-            |  #{formatted_location new_block}
-            |The new definition will overwrite the original one.
-          WARNING
+          old_definition = formatted_location existing_module.definition
+          new_definition = formatted_location new_block
+
+          if old_definition == new_definition
```

Comment:
Thanks for the feedback! I've addressed your comment and hopefully I did something along the lines of what you were expecting. I figured it was still important that the definition of the shared example group to be on the same line in the same file, but I've added a check to ensure that the file was loaded by RSpec as well so we now know that's the root cause of the issue. I'd love your feedback on my instructions for resolution in the warning message, though. I think they're clear, but they may not be the best instructions for the user, so your input would be much appreciated.